src/qmmm_gpw_forces.F

```diff
@@ -216,10 +217,28 @@ SUBROUTINE qmmm_forces(qs_env, qmmm_env, mm_particles, calc_force, mm_cell)
                CALL auxbas_pool%create_pw(rho_tot_r2)
                CALL pw_transfer(rho0_s_gs, rho_tot_r2)
                CALL pw_axpy(rho_tot_r2, rho_tot_r)
+               IF (ASSOCIATED(rhoz_cneo_s_gs)) THEN
+                  NULLIFY (rho_tot_r3)
+                  ALLOCATE (rho_tot_r3)
```

Comment:
rho_tot_r3 need not be a POINTER anymore.

Reply:
I was copying the code above, which used `rho_tot_r` and `rho_tot_r2` and they are pointers. `rho_tot_r3` is a similar temporary variable that I created to avoid reusing a not-yet-released pointer. I think it should still be a pointer? Otherwise the others should be changed too.

Reply:
The routines with which you call `rho_tot_r3`, do not require pointers and the variable is only allocated within the branches. The pools do not require points because they copy the PW grid objects themselves (which is only a few pointers). The code you copy-pasted indeed does not need pointers although it requires a simple modification of two routines. I have already done some cleanup but apparently missed a few spots.

Reply:
I removed the pointer attributes of `rho_tot_r` and `rho_tot_r2` in the master branch.